core/src/main/java/CoroutinesGatt.kt

```diff
@@ -26,14 +26,19 @@ import com.juul.able.experimental.messenger.OnConnectionStateChange
 import com.juul.able.experimental.messenger.OnDescriptorWrite
 import com.juul.able.experimental.messenger.OnMtuChanged
 import kotlinx.coroutines.experimental.CompletableDeferred
+import kotlinx.coroutines.experimental.Job
 import kotlinx.coroutines.experimental.channels.BroadcastChannel
 import java.util.UUID
+import kotlin.coroutines.experimental.CoroutineContext
 
 class CoroutinesGatt(
     private val bluetoothGatt: BluetoothGatt,
     private val messenger: Messenger
 ) : Gatt {
 
+    private val job = Job()
+    override val coroutineContext: CoroutineContext = job
```

Comment:
No need to cache a field for a virtual method. Should use `get() = job` instead of assignment here.

Reply:
Good point.

Per your suggestion, the JVM would probably optimize out the `coroutineContext` getter that pointed to the `job` anyways.

I did it more out of habit, IMHO holding the extra reference becomes valuable when Coroutine contexts are being combined or dispatchers being applied, e.g.:

```kotlin
    private val job = Job()
    override val coroutineContext: CoroutineContext = job + Dispatchers.Main
```

Since the `+` operator on `CoroutineContext` is non-trivial I assume that the memory expense of storing it is worth it for the savings in recomputing it on every invocation of the `coroutineContext` getter?